src/io/XTcpSocket.cpp

```diff
@@ -0,0 +1,193 @@
+#include "XTcpSocket.h"
+
+// platform dependent system headers
+#ifdef _MSC_VER
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#define SHUT_RDWR SD_BOTH
+#define MSG_NOSIGNAL 0
+#else
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+#endif
+
+// platform independent system headers
+#include <errno.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+void XTcpSocket::FormatLastErrorStr(const char* func) {
+#ifdef _MSC_VER
+    int err;
+
+    char msgbuf[256];  // for a message up to 255 bytes.
+
```

Comment:
please remove all of these extra newlines

Reply:
Removed.